docs/content/docs/tasks_usage/traffic_management/ingress.md

```diff
@@ -28,7 +28,13 @@ Ingress allows you to easily establish rules for traffic routing without creatin
 
 ## What types of ingress are supported by OSM v0.8.0?
 
-Currently, OSM v0.8.0 supports HTTP ingress. HTTPS ingress support is experimental, with support for one-way TLS authentication to backend services.
+Currently, OSM supports HTTP ingress. HTTPS ingress support is experimental, with support for one-way TLS authentication to backend services.
+
+## Supported Kubernetes Ingress API versions
+
+Since OSM only supports Kubernetes cluster versions >= v1.18.0, the API version for the Kubernetes Ingress resource must be precisely one of `networking.k8s.io/v1` or `networking.k8s.io/v1beta1`. OSM controller dynamically negotiates the Ingress API versions served by the Kubernetes API server and enables the same versions to be served in OSM.
+
+> Note: If either of these versions are not served by the Kubernetes API server for some reason, OSM controller will exit on failing to initialize its ingress client. This likely indicates an unsupported Kubernetes version on the cluster.
```

Comment:
i guess we will always find one to server, but also allowing OSM to keep working when no ingress resource kind is served would be not too hard either

Reply:
That is correct, but we do need the ingress client to initialize during the initialization phase of osm-controller. Silently ignoring no versions being present indicates perhaps unsupported k8s version, as noted in the doc, which needs to be caught during init phase.